src/test/payment.rs

```diff
@@ -63,10 +63,18 @@ async fn success() {
     assert_eq!(payment.asset_id, Some(asset_id.clone()));
     assert_eq!(payment.asset_amount, asset_amount);
     assert_eq!(payment.status, HTLCStatus::Succeeded);
+    assert!(
+        payment.preimage.is_some(),
+        "Payment preimage should be present for successful payment"
+    );
```

Comment:
```suggestion
    let payment_preimage_hash = hex_str(
        &Sha256::hash(&hex_str_to_vec(&payment.preimage.unwrap()).unwrap()).to_byte_array(),
    );
    assert_eq!(payment_preimage_hash, decoded.payment_hash);
```

please do this in this entire test, instead of checking that it's some. 

you will also need to modify the top of the file like this:
```diff
+use bitcoin::hashes::sha256::Hash as Sha256;
+use bitcoin::hashes::Hash;
+
 use crate::routes::{BitcoinNetwork, TransactionType, TransferKind, TransferStatus};
+use crate::utils::hex_str;
```